src/actions/SettingsActions.tsx

```diff
@@ -247,7 +247,7 @@ export function showUnlockSettingsModal(): ThunkAction<Promise<void>> {
   }
 }
 
-export function showRestoreWalletsModal(navigation: NavigationBase): ThunkAction<Promise<void>> {
+export function showRestoreWalletsModal(navigation: NavigationBase, restoreKeys: EdgeWalletInfoFull[]): ThunkAction<Promise<void>> {
```

Comment:
Can we remove this action, and just put the logic on the scene? That is the only place where this logic is used. The scene should really be in charge of its own UX, including this modal.

Also, the `Promise.all` down below is a bit silly, because `account.changeWalletStates` can handle multiple keys at once. When you copy the code over, do this instead:

```ts
const states: EdgeWalletStates ={}
for (const info of restoreKeys) {
  states[info.id] = { archived: false, deleted: false }
}
await account.changeWalletStates(states)
```

Now we only call the core once, and the core only needs to access the disk once. Much faster.